### PR TITLE
[FIX] web: remove default_ from parent record

### PR DESCRIPTION
**Pull request by adwid**
When having a subview for a 2many field, if an onchange is triggered for
that field, the default values of the parent are still in the request
context.

To reproduce the issue:
(Need helpdesk_repair. Use demo data)
1. Helpdesk > Configuration > Helpdesk Teams, edit "Customer Care":
    - Enable "Repairs"
2. Create a storable and tracked-by-usn product P
3. Update the on-hand quantity:
    - 1 x P with serial S
4. Helpdesk > Customer Care (Tickets), create a ticket:
    - Product: P
    - Lot/Serial number: S
5. Click on Repair and save the repair order R
6. Edit R and add a line in "Parts"

Error: on the new line, both Product and Lot/Serial fields are already
filled with values P and S, which does not make sens

On step 5, when clicking on Repair, the form views of a repair order is
loaded with a specific context:
https://github.com/odoo/enterprise/blob/deef3c967d43862edf02dbcfc461fb89b064731e/helpdesk_repair/views/helpdesk_views.xml#L31-L34

Therefore, when editing the form and trying to add a new line in Parts,
an onchange is triggered but the default values are not removed from the
context. As a result, since the model `repair.line` also has two fields
called `product_id` and `lot_id`, the default values of the parent
(`repair.order`) are used.

When computing the context of a subview, the default values of the
parent should be removed.

OPW-2752436

### CODE REVIEW

**Comment by adwid** [comment]
Hi @ged-odoo 
Here is the issue we talked about on Monday
Can you review the fix ?

**Comment by ged-odoo** [comment]
@aab-odoo

File: addons/web/static/tests/views/form_tests.js [hunk 1]
```diff
@@ -2009,6 +2009,41 @@ QUnit.module('Views', {
...
+                context: {default_state: "ab"}
+            },
+            arch: '<form string="Partners">' +
+                    '<field name="product_ids" context="{\'default_product_uom_qty\': 68}">' +
+                      '<tree editable="top">' +
+                        '<field name="name"/>' +
+                      '</tree>' +
+                    '</field>' +
+                  '</form>',
+            res_id: 1,
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js:2032]: remove the res_id here

File: addons/web/static/tests/views/form_tests.js [hunk 2]
```diff
@@ -2009,6 +2009,41 @@ QUnit.module('Views', {
...
+            mockRPC: function (route, args) {
+                if (route === "/web/dataset/call_kw/product/onchange") {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js:2034]: without the res_id (see above), you'll pass twice here: once for the main record, and once for the x2many record. You can then assert that the context of the first call contains default_state (only) and the second one contains default_product_uom_qty (only).

**Review by aab-odoo — APPROVED** [review]

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**